src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -232,14 +235,63 @@ protected void onStartup() {
         super.onStartup();
 
         initializeReference();
+
         initializeReads();
+
         initializeFeatures();
+
         initializeIntervals(); // Must be initialized last, since intervals currently require a sequence dictionary from another data source
 
+        validateSequenceDictionaries();
+
         checkToolRequirements();
     }
 
     /**
+     * Validate all sequence dictionaries by checking them against each other.
+     */
+    private void validateSequenceDictionaries() {
+        SAMSequenceDictionary refDict = null;
+        if (hasReference()){
+            refDict = reference.getSequenceDictionary();
+        }
+        SAMSequenceDictionary readDict = null;
+        if (hasReads()) {
+            readDict = reads.getSequenceDictionary();
+        }
+        List<SAMSequenceDictionary> featureDictList = new ArrayList<>();
+        if (hasFeatures()){
+            List<VCFHeader> featureHeaders = features.getAllVariantHeaders();
+            for (VCFHeader header : featureHeaders) {
+                SAMSequenceDictionary headerDict = header.getSequenceDictionary();
+                if (headerDict != null) {
+                    featureDictList.add(headerDict);
+                }
+            }
+        }
+        //check reference dict against reads dict
+        if (hasReference() && hasReads()) {
+            SequenceDictionaryUtils.validateDictionaries("reference", refDict, "reads", readDict, true, null);
+        }
+
+        //check any feature dicts against the ref or read dict
+        for (int k = 0; k < featureDictList.size(); k++){
+            if (hasReference()){
+                SequenceDictionaryUtils.validateDictionaries("reference", refDict, "feature", featureDictList.get(k), false, null);
+            }
+            if (hasReads()) {
+                SequenceDictionaryUtils.validateDictionaries("reads", readDict, "feature", featureDictList.get(k), false, null);
+            }
+            if (k+1 < featureDictList.size()) {
+                SequenceDictionaryUtils.validateDictionaries("feature1", featureDictList.get(k), "feature2", featureDictList.get(k+1), false, null);
```

Comment:
Create a new ticket to modify this method to pass in the vcf file names associated with each sequence dictionary into `validateDictionaries()` (instead of generic names like "feature1") to make error messages more useful.